pilot/cmd/pilot-discovery/main.go

```diff
@@ -149,6 +149,8 @@ func init() {
 	if err := discoveryCmd.PersistentFlags().MarkDeprecated("appNamespace", "please use ${APP_NAMESPACE} environment variable instead"); err != nil {
 		panic(err)
 	}
+	discoveryCmd.PersistentFlags().BoolVarP(&serverArgs.Config.ControllerOptions.EnableNodePortGateways, "enableNodePortGateways", "",
+		false, "Whether to watch nodes and advertise their addresses as endpoints for gateways with NodePort services")
```

Comment:
You could have it `true` by default - so that it doesn't change current behavior - and we could set that value to false in our charts. Thus you can upstream this patch.

Reply:
Sure. I wasn't really sure upstream would be interested in it. It feels like whack-a-mole until there is any kind of policy around cluster-scope permissions upstream, but we can try.

Reply:
Updated.